src/FreeAge/common/resources.hpp

```diff
@@ -51,6 +53,21 @@ struct ResourceAmount {
     }
     return true;
   }
+
+  /// Returns the number of time that the given resource amount can be subtracted from
+  /// this resource amount.
+  /// TODO: rename ?
+  inline int CanAffordTimes(const ResourceAmount& other) const {
```

Comment:
I have found this kind of function on a collection of item very useful in many situation, I usually call it `has` but it doesn't seem to fit with this codebase...